### PR TITLE
PPLUS-105: Basic structure + Client abstraction

### PR DESCRIPTION
**Pull request by alexanderM91**
### Overview
- Ticket: https://spryker.atlassian.net/browse/PPLUS-105

### Change log
- Introduced `upgrader upgrade` command. 

### Note:
The command contains: 
* checks, if exist any uncommitted changes and return error "Please commit or revert your changes" when exists.
* run "composer update" command after checks.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    "extra": {
+        "branch-alias": {
+            "dev-master": "1.0.x-dev"
+        }
+    },
+    "config": {
+        "sort-packages": true
+    },
+    "scripts": {
+        "stan": "phpstan analyze -l 5 src/",
```
Comment (dereuromark) [inline composer.json:40]:
```suggestion
        "stan": "phpstan analyze",
```
as you neon file contains already everything.

File: src/Upgrader/Business/UpgraderFacadeInterface.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ */
+
+namespace Upgrader\Business;
+
+use Upgrader\Business\Command\ResultOutput\CommandResultOutput;
+
+interface UpgraderFacadeInterface
+{
+    /**
+     * Specification:
```
Comment (alexanderM91) [inline src/Upgrader/Business/UpgraderFacadeInterface.php:15]: Please add a specification

File: src/Upgrader/Business/PackageManager/Client/Composer/Json/Writer/ComposerJsonWriter.php [hunk 3]
```diff
@@ -89,4 +89,12 @@ protected function adjustIndentation(string $encodedJson, int $indentation): str
     {
         return $this->printer->print($encodedJson, str_repeat(' ', $indentation));
     }
+
+    /**
+     * @return string
+     */
+    protected function getFileName(): string
```
Comment (dereuromark) [inline src/Upgrader/Business/PackageManager/Client/Composer/Json/Writer/ComposerJsonWriter.php:96]: Why is this needed?

File: src/Upgrader/Business/UpgraderFacadeInterface.php (hunk 2 above)
Comment (VladislavStrelchenko) [reply src/Upgrader/Business/UpgraderFacadeInterface.php:15]: Added ++ 
Actual PR: https://github.com/spryker-sdk/upgrader/pull/2

File: src/Upgrader/Business/PackageManager/Client/Composer/Json/Writer/ComposerJsonWriter.php (hunk 3 above)
Comment (VladislavStrelchenko) [reply src/Upgrader/Business/PackageManager/Client/Composer/Json/Writer/ComposerJsonWriter.php:97]: For mock method in `ComposerJsonWriterTest`
Actual PR: https://github.com/spryker-sdk/upgrader/pull/2
